src/otx/core/data/entity/base.py

```diff
@@ -0,0 +1,157 @@
+"""Module for OTX base data entities."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from enum import IntEnum, auto
+from typing import Dict, Generic, TypeVar
+
+import numpy as np
+from torch import Tensor
+from torchvision import tv_tensors
+
+from otx.core.types.task import OTXTaskType
+
+
+@dataclass
+class ImageInfo:
+    """Meta info for image.
+
+    :param img_id: Image id
+    :param img_shape: Image shape after preprocessing
+    :param ori_shape: Image shape right after loading it
+    :param pad_shape: Image shape before padding
+    :param scale_factor: Scale factor if the image is rescaled during preprocessing
+    """
+
+    img_idx: int
+    img_shape: tuple[int, int]
+    ori_shape: tuple[int, int]
+    pad_shape: tuple[int, int]
+    scale_factor: tuple[float, float]
+
+
+class ImageType(IntEnum):
+    """Enum to indicate the image type in `ImageInfo` class."""
+
+    NUMPY = auto()
+    TV_IMAGE = auto()
+
+
+T_OTXDataEntity = TypeVar(
+    "T_OTXDataEntity",
+    bound="OTXDataEntity",
+)
+
+
+@dataclass
+class OTXDataEntity:
+    """Base data entity for OTX.
+
+    This entity is the output of each OTXDataset,
+    which can be go through the input preprocessing tranforms.
+
+    :param task: OTX task definition
+    :param image: Image tensor which can have different type according to `image_type`
+        1) `image_type=ImageType.NUMPY`: H x W x C numpy image tensor
+        2) `image_type=ImageType.TV_IMAGE`: C x H x W torchvision image tensor
+    :param imgs_info: Meta information for images
+    """
+
+    image: np.ndarray | tv_tensors.Image
+    img_info: ImageInfo
+
+    @property
+    def task(self) -> OTXTaskType:
```

Comment:
Why is there a need to specify a task for the `OTXDataEntity`? Given that it's a base data element, it could potentially be utilized across various tasks. So, what's the rationale behind binding it to a specific task?

Reply:
`OTXDataEntity` is a base class of other data entities. It makes that the other data entities should have task.

https://github.com/openvinotoolkit/training_extensions/blob/b8d8654ac4e846af0d0c4f1b0e971c5332ff6932/src/otx/core/data/entity/base.py#L64-L68

https://github.com/openvinotoolkit/training_extensions/blob/8c8c50f64571c4b25ddcf5a1276ebaba5a4bcca9/src/otx/core/data/entity/detection.py#L52-L55